### DISCUSSION TITLE
[Adonis V5] API Rate Limit in Adonis V5?

### DISCUSSION THREAD

**Discussion by aayush123** [adonisjs/core#1738] [Help]
Hi,

I was wondering if there's a way to rate limit API in Adonis V5? Any community packages or code snippets to point the way would be of great help.

Thanks!

**Comment by fairhat** [comment]
@aayush123 

I'd say A custom Middleware should do the job.
Since Rate limiting is an abstract term, what exactly do you want to limit and on which condition?
If you want to rate limit on specific route (groups), then create a middleware that you can apply to that group
If you want to limit in general, well then create one middleware and group all your routes with that middleware

If you want to customize the condition, maybe a custom Model could help.


APIUser
--
id
...
apiCalls: integer

So whenever a call is made, your middleware +1s the apiCalls property for that user and if it exceeds your limit, you block the call by throwing an exception
